executor/load_data.go

```diff
@@ -727,285 +853,164 @@ func (e *LoadDataWorker) commitWork(ctx context.Context) (err error) {
 	}
 	return nil
 }
 
-// ReadOneBatchRows reads rows from parser. When parser's reader meet EOF, it
-// will return nil. For other errors it will return directly. When the rows
-// batch is full it will also return nil.
-// The result rows are saved in e.rows and update some members, caller can check
-// if curBatchCnt == 0 to know if reached EOF.
-func (e *LoadDataWorker) ReadOneBatchRows(ctx context.Context, parser mydump.Parser) error {
-	for {
-		if err := parser.ReadRow(); err != nil {
-			if errors.Cause(err) == io.EOF {
-				return nil
-			}
-			return exeerrors.ErrLoadDataCantRead.GenWithStackByArgs(
-				err.Error(),
-				"Only the following formats delimited text file (csv, tsv), parquet, sql are supported. Please provide the valid source file(s)",
-			)
-		}
-		// rowCount will be used in fillRow(), last insert ID will be assigned according to the rowCount = 1.
-		// So should add first here.
-		e.rowCount++
-		r, err := e.parserData2TableData(ctx, parser.LastRow().Row)
-		if err != nil {
-			return err
-		}
-		e.rows = append(e.rows, r)
-		e.curBatchCnt++
-		if e.maxRowsInBatch != 0 && e.rowCount%e.maxRowsInBatch == 0 {
-			logutil.Logger(ctx).Info("batch limit hit when inserting rows", zap.Int("maxBatchRows", e.maxChunkSize),
-				zap.Uint64("totalRows", e.rowCount))
-			return nil
-		}
-	}
-}
-
-// parserData2TableData encodes the data of parser output.
-func (e *LoadDataWorker) parserData2TableData(
-	ctx context.Context,
-	parserData []types.Datum,
-) ([]types.Datum, error) {
-	var errColNumMismatch error
-	switch {
-	case len(parserData) < e.controller.GetFieldCount():
-		errColNumMismatch = exeerrors.ErrWarnTooFewRecords.GenWithStackByArgs(e.rowCount)
-	case len(parserData) > e.controller.GetFieldCount():
-		errColNumMismatch = exeerrors.ErrWarnTooManyRecords.GenWithStackByArgs(e.rowCount)
-	}
-
-	if errColNumMismatch != nil {
-		if e.restrictive {
-			return nil, errColNumMismatch
-		}
-		e.handleWarning(errColNumMismatch)
-	}
-
-	row := make([]types.Datum, 0, len(e.insertColumns))
-	sessionVars := e.Ctx.GetSessionVars()
-	setVar := func(name string, col *types.Datum) {
-		// User variable names are not case-sensitive
-		// https://dev.mysql.com/doc/refman/8.0/en/user-variables.html
-		name = strings.ToLower(name)
-		if col == nil || col.IsNull() {
-			sessionVars.UnsetUserVar(name)
-		} else {
-			sessionVars.SetUserVarVal(name, *col)
-		}
-	}
-
-	fieldMappings := e.controller.GetFieldMapping()
-	for i := 0; i < len(fieldMappings); i++ {
-		if i >= len(parserData) {
-			if fieldMappings[i].Column == nil {
-				setVar(fieldMappings[i].UserVar.Name, nil)
-				continue
-			}
+// mergeAndSetMessage merges stats from all used session context and sets info
+// message(ERR_LOAD_INFO) generated by LOAD statement to UserSctx, it is
+// public for test.
+func (e *LoadDataWorker) mergeAndSetMessage() string {
+	encodeStmtCtx := e.encodeWorker.ctx.GetSessionVars().StmtCtx
+	numAffected := encodeStmtCtx.AffectedRows()
```

Comment:
is this always 0?

Reply:
Seems yes, I'll check CI without counting encodeStmtCtx.